homeassistant/components/gree/switch.py

```diff
@@ -41,162 +64,30 @@ def init_device(coordinator):
-        """Turn the entity on."""
-        self.coordinator.device.quiet = True
-        await self.coordinator.push_state_update()
-        self.async_write_ha_state()
-
-    async def async_turn_off(self, **kwargs: Any) -> None:
-        """Turn the entity off."""
-        self.coordinator.device.quiet = False
-        await self.coordinator.push_state_update()
-        self.async_write_ha_state()
-
-
-class GreeFreshAirSwitchEntity(GreeEntity, SwitchEntity):
-    """Representation of the fresh air mode state of the device."""
-
-    def __init__(self, coordinator):
-        """Initialize the Gree device."""
-        super().__init__(coordinator, "Fresh Air")
+class GreeSwitch(GreeEntity, SwitchEntity):
+    """Generic Gree switch entity."""
 
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return SwitchDeviceClass.SWITCH
+    _attr_device_class = SwitchDeviceClass.SWITCH
 
-    @property
-    def is_on(self) -> bool:
-        """Return if the state is turned on."""
-        return self.coordinator.device.fresh_air
-
-    async def async_turn_on(self, **kwargs: Any) -> None:
-        """Turn the entity on."""
-        self.coordinator.device.fresh_air = True
-        await self.coordinator.push_state_update()
-        self.async_write_ha_state()
-
-    async def async_turn_off(self, **kwargs: Any) -> None:
-        """Turn the entity off."""
-        self.coordinator.device.fresh_air = False
-        await self.coordinator.push_state_update()
-        self.async_write_ha_state()
-
-
-class GreeXFanSwitchEntity(GreeEntity, SwitchEntity):
-    """Representation of the extra fan mode state of the device."""
-
-    def __init__(self, coordinator):
+    def __init__(self, coordinator, description: SwitchEntityDescription) -> None:
         """Initialize the Gree device."""
-        super().__init__(coordinator, "XFan")
+        self.entity_description = description
 
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return SwitchDeviceClass.SWITCH
+        super().__init__(coordinator, cast(str, description.name))
 
     @property
     def is_on(self) -> bool:
         """Return if the state is turned on."""
-        return self.coordinator.device.xfan
+        return getattr(self.coordinator.device, self.entity_description.key)
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the entity on."""
-        self.coordinator.device.xfan = True
+        setattr(self.coordinator.device, self.entity_description.key, True)
         await self.coordinator.push_state_update()
         self.async_write_ha_state()
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the entity off."""
-        self.coordinator.device.xfan = False
+        setattr(self.coordinator.device, self.entity_description.key, False)
```

Comment:
Avoid the use of `setattr` here, adding a lambda to the entity description might be more fitting (and maintains typing).